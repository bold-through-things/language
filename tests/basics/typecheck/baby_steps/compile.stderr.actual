[
  {
    "recoverable": false,
    "line": 25,
    "char": 0,
    "content": "local test",
    "error": "failed to assert: field demands int but is given str",
    "error_type": "FIELD_TYPE_MISMATCH"
  },
  {
    "recoverable": false,
    "line": 30,
    "char": 0,
    "content": "local test_str",
    "error": "failed to assert: field demands str but is given None",
    "error_type": "MISSING_TYPE"
  },
  {
    "recoverable": false,
    "line": 34,
    "char": 0,
    "content": "local test_int",
    "error": "failed to assert: field demands int but is given str",
    "error_type": "FIELD_TYPE_MISMATCH"
  },
  {
    "recoverable": false,
    "line": 38,
    "char": 0,
    "content": "67lang:access_local test_int",
    "error": "failed to assert: field demands int but is given str",
    "error_type": "FIELD_TYPE_MISMATCH"
  },
  {
    "recoverable": false,
    "line": 42,
    "char": 0,
    "content": "67lang:access_local test_str",
    "error": "failed to assert: field demands str but is given int",
    "error_type": "FIELD_TYPE_MISMATCH"
  },
  {
    "recoverable": false,
    "line": 45,
    "char": 0,
    "content": "local test_bool",
    "error": "failed to assert: field demands bool but is given str",
    "error_type": "FIELD_TYPE_MISMATCH"
  },
  {
    "recoverable": false,
    "line": 49,
    "char": 0,
    "content": "67lang:access_local test_bool",
    "error": "failed to assert: field demands bool but is given str",
    "error_type": "FIELD_TYPE_MISMATCH"
  },
  {
    "recoverable": false,
    "line": 52,
    "char": 0,
    "content": "67lang:call split",
    "error": "failed to assert: argument demands str and is given bool",
    "error_type": "ARGUMENT_TYPE_MISMATCH"
  },
  {
    "recoverable": false,
    "line": 54,
    "char": 0,
    "content": "67lang:call split",
    "error": "failed to assert: argument demands str and is given list",
    "error_type": "ARGUMENT_TYPE_MISMATCH"
  },
  {
    "recoverable": false,
    "line": 49,
    "char": 0,
    "content": "67lang:access_local test_bool",
    "error": "failed to assert: single child used for assignment",
    "error_type": "ASSERTION_FAILED"
  }
]
