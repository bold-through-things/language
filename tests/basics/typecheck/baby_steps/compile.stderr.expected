[
  {
    "recoverable": false,
    "line": 25,
    "char": 0,
    "content": "local test",
    "error": "failed to assert: field demands int but is given str",
    "error_type": "FIELD_TYPE_MISMATCH"
  },
  {
    "recoverable": false,
    "line": 30,
    "char": 0,
    "content": "local test_str",
    "error": "failed to assert: field demands str but is given None",
    "error_type": "MISSING_TYPE"
  },
  {
    "recoverable": false,
    "line": 34,
    "char": 0,
    "content": "local test_int",
    "error": "failed to assert: field demands int but is given str",
    "error_type": "FIELD_TYPE_MISMATCH"
  },
  {
    "recoverable": false,
    "line": 38,
    "char": 0,
    "content": "67lang:call test_int",
    "error": "failed to assert: could not find a matching overload for test_int with arguments ['str']",
    "error_type": "NO_MATCHING_OVERLOAD",
    "visible_overloads": [
      {
        "fn": "_0x13_test_int",
        "demands": [],
        "returns": "int",
        "convention": "LocalAccessCall"
      },
      {
        "fn": "_0x13_test_int",
        "demands": [
          "int"
        ],
        "returns": "int",
        "convention": "LocalAccessCall"
      }
    ]
  },
  {
    "recoverable": false,
    "line": 42,
    "char": 0,
    "content": "67lang:call test_str",
    "error": "failed to assert: could not find a matching overload for test_str with arguments ['int']",
    "error_type": "NO_MATCHING_OVERLOAD",
    "visible_overloads": [
      {
        "fn": "_0x10_test_str",
        "demands": [],
        "returns": "str",
        "convention": "LocalAccessCall"
      },
      {
        "fn": "_0x10_test_str",
        "demands": [
          "str"
        ],
        "returns": "str",
        "convention": "LocalAccessCall"
      }
    ]
  },
  {
    "recoverable": false,
    "line": 45,
    "char": 0,
    "content": "local test_bool",
    "error": "failed to assert: field demands bool but is given str",
    "error_type": "FIELD_TYPE_MISMATCH"
  },
  {
    "recoverable": false,
    "line": 50,
    "char": 0,
    "content": "67lang:call test_bool",
    "error": "failed to assert: could not find a matching overload for test_bool with arguments ['str']",
    "error_type": "NO_MATCHING_OVERLOAD",
    "visible_overloads": [
      {
        "fn": "_0x1c_test_bool",
        "demands": [],
        "returns": "bool",
        "convention": "LocalAccessCall"
      },
      {
        "fn": "_0x1c_test_bool",
        "demands": [
          "bool"
        ],
        "returns": "bool",
        "convention": "LocalAccessCall"
      }
    ]
  },
  {
    "recoverable": false,
    "line": 53,
    "char": 0,
    "content": "67lang:call split",
    "error": "failed to assert: could not find a matching overload for split with arguments ['bool']",
    "error_type": "NO_MATCHING_OVERLOAD",
    "visible_overloads": [
      {
        "field": "split",
        "demands": [
          "SymbolConstructor"
        ],
        "returns": "uniquesymbol",
        "convention": "FieldCall"
      },
      {
        "constructor": "String",
        "fn": "split",
        "demands": [
          "str"
        ],
        "returns": "list",
        "convention": "PrototypeCall"
      },
      {
        "constructor": "String",
        "fn": "split",
        "demands": [
          "str",
          "regex"
        ],
        "returns": "list",
        "convention": "PrototypeCall"
      },
      {
        "constructor": "String",
        "fn": "split",
        "demands": [
          "str",
          "regex",
          "float"
        ],
        "returns": "list",
        "convention": "PrototypeCall"
      },
      {
        "constructor": "String",
        "fn": "split",
        "demands": [
          "str",
          "str"
        ],
        "returns": "list",
        "convention": "PrototypeCall"
      },
      {
        "constructor": "String",
        "fn": "split",
        "demands": [
          "str",
          "str",
          "float"
        ],
        "returns": "list",
        "convention": "PrototypeCall"
      }
    ]
  },
  {
    "recoverable": false,
    "line": 56,
    "char": 0,
    "content": "67lang:call split",
    "error": "failed to assert: could not find a matching overload for split with arguments ['list']",
    "error_type": "NO_MATCHING_OVERLOAD",
    "visible_overloads": [
      {
        "field": "split",
        "demands": [
          "SymbolConstructor"
        ],
        "returns": "uniquesymbol",
        "convention": "FieldCall"
      },
      {
        "constructor": "String",
        "fn": "split",
        "demands": [
          "str"
        ],
        "returns": "list",
        "convention": "PrototypeCall"
      },
      {
        "constructor": "String",
        "fn": "split",
        "demands": [
          "str",
          "regex"
        ],
        "returns": "list",
        "convention": "PrototypeCall"
      },
      {
        "constructor": "String",
        "fn": "split",
        "demands": [
          "str",
          "regex",
          "float"
        ],
        "returns": "list",
        "convention": "PrototypeCall"
      },
      {
        "constructor": "String",
        "fn": "split",
        "demands": [
          "str",
          "str"
        ],
        "returns": "list",
        "convention": "PrototypeCall"
      },
      {
        "constructor": "String",
        "fn": "split",
        "demands": [
          "str",
          "str",
          "float"
        ],
        "returns": "list",
        "convention": "PrototypeCall"
      }
    ]
  }
]
